core/unit_test/cuda/TestCuda_InterOp_StreamsMultiGPU.cpp

```diff
@@ -106,36 +106,23 @@ TEST(cuda_multi_gpu, scratch_space) {
   }
 }
 
-// Test that stream synchronization behavoir for CudaAPI matches the assumptions
-// made in Kokkos for multi gpu support, namely, that any stream (no matter
-// which device it is created on) can be synced from any device.
 TEST(cuda_multi_gpu, stream_sync_semantics) {
   StreamsAndDevices streams_and_devices;
   {
-    auto streams = streams_and_devices.streams;
+    std::array<TEST_EXECSPACE, 2> execs =
+        get_execution_spaces(streams_and_devices);
 
-    // Allocate data.
     int *value;
     int *check;
     KOKKOS_IMPL_CUDA_SAFE_CALL(
         cudaMallocHost(reinterpret_cast<void **>(&value), 1 * sizeof(int)));
     KOKKOS_IMPL_CUDA_SAFE_CALL(
         cudaMallocHost(reinterpret_cast<void **>(&check), 1 * sizeof(int)));
```

Comment:
I think this would need to be changed for the test to be able to run on any backend (in the future)